Create maestro-integration.md (#1431)
codemagic-ci-cd/codemagic-docs: content/integrations/maestro-integration.md

---
title: Maestro integration
description: How to integrate your workflows with Maestro using codemagic.yaml
weight: 14
---

> [**Maestro UI testing framework**](https://mobile.dev/) from `Mobile.dev` enables you to test your iOS and Android mobile apps using simply created tests flows steps.

Reviewer: ```suggestion
[**Maestro UI testing framework**](https://mobile.dev/) from `Mobile.dev` lets you test your iOS and Android mobile apps using simple to-create test flows.
```

{{<markdown>}}
See how to build your native android app [here](../yaml-quick-start/building-a-native-android-app/) or your Flutter app [here](../yaml-quick-start/building-a-flutter-app/).

Add the following script to your `publishing` section:
{{< highlight yaml "style=paraiso-dark">}}
publishing:
    scripts:
    - name: Run tests on Maestro cloud
        script: | 
        export PATH="$PATH":"$HOME/.maestro/bin"
        apkPath="/build/app/outputs/apk/release/app-release.apk"
        maestro cloud \
        --apiKey $MDEV_API_KEY \
        $apkPath \
        .maestro/
{{< /highlight >}}

Don't forget to change the value of the `apkPath` to your actual apk path.
{{</markdown>}}
{{< /tab >}}

{{< tab header="iOS" >}}
{{<markdown>}}
For iOS, you need to upload your x86-compatible Simulator `.app` directory.

Here's the script on how you can build it.
{{< highlight yaml "style=paraiso-dark">}}
  scripts:
    - name: Build unsigned .app
    script: | 
        xcodebuild \
        -workspace "ios/$XCODE_WORKSPACE" \
        -scheme "$XCODE_SCHEME" \
        -configuration "Debug" \
        -sdk iphonesimulator \
        -derivedDataPath ios/output
{{< /highlight >}}

> Don't forget to add the environment variables that holds your XCode workspace name under `$XCODE_WORKSPACE` and the Scheme name under `$XCODE_SCHEME`, see the complete sample project from [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/tree/main/integrations/maestro_integration_demo_project/codemagic.yaml).

Reviewer: ```suggestion
Don't forget to add the environment variables that holds your XCode workspace name under `$XCODE_WORKSPACE` and the Scheme name under `$XCODE_SCHEME`. See the complete sample project [here](https://github.com/codemagic-ci-cd/codemagic-sample-projects/tree/main/integrations/maestro_integration_demo_project/codemagic.yaml).
```

{{</markdown>}}
{{< /tab >}}
{{< /tabpane >}}

> That's it, now if your Codemagic's build has failed at the maestro cloud step, then your tests has failed, or which means everything went well, and you can check out the build page for more details.

Reviewer: ```suggestion
If your Codemagic's build has failed at the Maestro cloud step, then your tests have failed. Otherwise, everything went well and you can check out the build page for more details.
```
